### PR TITLE
[Update `rna-seq-expression-summary-stats` module for v7 Part 1] use `annotator` API

### PR DESCRIPTION
**Pull request by logstar**
### Purpose/implementation Section

In the `rna-seq-expression-summary-stats` module, add RMTL, EFO, and MONDO annotation columns to the long-format tables using [the `annotator` API](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/long-format-table-utils).

#### What was your approach?

In commit <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/commit/ad28e3ec9372bd13a2196bf13eb5577d523a0955>:

- Removed code that was used for adding annotation columns, in `01-tpm-summary-stats.R`.
- Added code for using the `annotator` API to add annotation columns, in `01-tpm-summary-stats.R`.
- Rerun the `rna-seq-expression-summary-stats` module.
- Confirmed that the results are identical to the previously committed results.

In commits <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/commit/78b77e430b9e0010bf70cc95ee4e6c15980c09aa>, <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/commit/d6862ce6e8e582d74f751fd2d11ab8d01856020c>, and <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/commit/752549b8415ccf48f1487ad78a76b95509fb88f7>:

- Added a unit testing framework using the [`testthat`](https://testthat.r-lib.org/index.html) package to the `rna-seq-expression-summary-stats` module, in order to remove test case comments in `01-tpm-summary-stats.R`.

#### What GitHub issue does your pull request address?

<https://github.com/PediatricOpenTargets/ticket-tracker/issues/122>

#### Which areas should receive a particularly close look?

Changes in `01-tpm-summary-stats.R`.

#### Is there anything that you want to discuss further?

(EDIT 07/26/2021 YZ)

The results are generated using v6 data release and independent sample lists, even though [the v7 independent sample lists were merged](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/62) and [the v7 data release was merged](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/61), in order to demonstrate that the results are identical to previous results after the refactoring.

I created a ticket to rerun the `rna-seq-expression-summary-stats` module using v7 data release, <https://github.com/PediatricOpenTargets/ticket-tracker/issues/130>. The updates are in PR <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/67>.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

Tables.

#### What is your summary of the results?

```text
$ ls -lh results/
total 358M
-rw-r--r-- 1 ubuntu ubuntu 8.5M Jul 25 20:47 cancer_group_all_cohort_mean_tpm.tsv
-rw-r--r-- 1 ubuntu ubuntu  21M Jul 25 20:47 cancer_group_all_cohort_mean_tpm_cancer_group_wise_quantiles.tsv
-rw-r--r-- 1 ubuntu ubuntu  13M Jul 25 20:47 cancer_group_all_cohort_mean_tpm_cancer_group_wise_z_scores.tsv
-rw-r--r-- 1 ubuntu ubuntu  12M Jul 25 20:47 cancer_group_all_cohort_mean_tpm_gene_wise_z_scores.tsv
-rw-r--r-- 1 ubuntu ubuntu  12K Jul 25 20:45 cancer_group_all_cohort_sample_metadata.tsv
-rw-r--r-- 1 ubuntu ubuntu 9.5M Jul 25 20:47 cancer_group_all_cohort_standard_deviation_tpm.tsv
-rw-r--r-- 1 ubuntu ubuntu 8.5M Jul 25 20:48 cancer_group_individual_cohort_mean_tpm.tsv
-rw-r--r-- 1 ubuntu ubuntu  21M Jul 25 20:48 cancer_group_individual_cohort_mean_tpm_cancer_group_wise_quantiles.tsv
-rw-r--r-- 1 ubuntu ubuntu  13M Jul 25 20:48 cancer_group_individual_cohort_mean_tpm_cancer_group_wise_z_scores.tsv
-rw-r--r-- 1 ubuntu ubuntu  12M Jul 25 20:49 cancer_group_individual_cohort_mean_tpm_gene_wise_z_scores.tsv
-rw-r--r-- 1 ubuntu ubuntu  11K Jul 25 20:47 cancer_group_individual_cohort_sample_metadata.tsv
-rw-r--r-- 1 ubuntu ubuntu 9.5M Jul 25 20:48 cancer_group_individual_cohort_standard_deviation_tpm.tsv
-rw-r--r-- 1 ubuntu ubuntu  43M Jul 25 20:49 long_n_tpm_mean_sd_quantile_gene_wise_zscore.json.gz
-rw-r--r-- 1 ubuntu ubuntu  43M Jul 25 20:50 long_n_tpm_mean_sd_quantile_gene_wise_zscore.jsonl.gz
-rw-r--r-- 1 ubuntu ubuntu  37M Jul 25 20:49 long_n_tpm_mean_sd_quantile_gene_wise_zscore.tsv.gz
-rw-r--r-- 1 ubuntu ubuntu  40M Jul 25 20:49 long_n_tpm_mean_sd_quantile_group_wise_zscore.json.gz
-rw-r--r-- 1 ubuntu ubuntu  40M Jul 25 20:50 long_n_tpm_mean_sd_quantile_group_wise_zscore.jsonl.gz
-rw-r--r-- 1 ubuntu ubuntu  34M Jul 25 20:49 long_n_tpm_mean_sd_quantile_group_wise_zscore.tsv.gz
```

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [NA, as CI is not available] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Review by afarrel — APPROVED** [review]
Reviewed changes. I don't see any issues.
Tested code line by line locally. Seems to work fine.

**Comment by logstar** [comment]
> Reviewed changes. I don't see any issues.
> Tested code line by line locally. Seems to work fine.

@afarrel Thank you for reviewing and test-running this PR!

**Comment by logstar** [comment]
@jharenza @afarrel I will merge this PR shortly. Before merging, I will not merge the latest dev branch to this PR, because this PR relies on v6 data and independent sample lists.

**Pull request merged**
